stan/math/prim/arr/functor/mpi_command.hpp

```diff
@@ -0,0 +1,52 @@
+#ifdef STAN_MPI
+
+#ifndef STAN_MATH_PRIM_ARR_FUNCTOR_MPI_COMMAND_HPP
+#define STAN_MATH_PRIM_ARR_FUNCTOR_MPI_COMMAND_HPP
+
+#include <boost/serialization/serialization.hpp>
+#include <boost/serialization/access.hpp>
+#include <boost/serialization/base_object.hpp>
+#include <boost/serialization/export.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * A MPI command object is used to execute code on worker nodes.
+ * The base class for all MPI commands defines the virtual run
+ * method. All MPI commands must inherit from this class and
+ * implement the run method which defines what will be executed on
+ * each remote.
+ *
+ * <p>Note that a concrete command must also register itself with
+ * the serialization boost library. A convenience macro
+ * STAN_REGISTER_MPI_COMMAND is provided which does enable
+ * additional tuning options for optimized MPI communication
+ * performance (marking object as trivially serializable and
+ * disable version tracking).
+ *
+ * For each declared command a call to the
+ * STAN_REGISTER_MPI_COMMAND is required. This macro must be
+ * called in the root namespace.
+ */
+struct mpi_command {
+  friend class boost::serialization::access;
+  template <class Archive>
+  void serialize(Archive& ar, const unsigned int version) {}
```

Comment:
It would be nice to doc these methods on the base class here.  That includes the template parameter `Archive`.

Presumably `ar` isn't constant because that's what's being written into.

Reply:
This is standard code required by the boost serialization library. I understand it only conceptually as I have just followed the doc. How about I doc that this is standard serialization code and add a reference to where to dig in?